products/views.py

```diff
@@ -2,28 +2,30 @@
 
 from django.http           import JsonResponse
 from django.views          import View
+from django.db.models      import Q
 
-from products.models  import Product, ProductImage, TasteByProduct, Grainding, Size
+from products.models       import Product, ProductImage, TasteByProduct, Grainding, Size
 
+from urllib.parse          import unquote
 
-class MainProductView(View): 
+
+class MainProductView(View):
     def get(self, request): 
         premiums             = Product.objects.all().order_by('-price')[:3]
         fresh_products       = Product.objects.all().order_by('-roasting_date')[:4]
-
         result_premium       = [{
                     'id'             : premium.id,
                     'name'           : premium.name,
                     'eng_name'       : premium.eng_name,
                     'img'            : [{
                         'img_id'     : image.id,
                         'img_url'    : image.url
-                    } for image in ProductImage.objects.filter(product_id = premium.id)],
+                    } for image in ProductImage.objects.select_related('product').filter(product_id = premium.id)],
                     'roasting_date'  : premium.roasting_date,
                     'taste'          : [{
                         'taste_id'   : flavor.taste.id,
                         'taste_name' : flavor.taste.name
-                    } for flavor in TasteByProduct.objects.filter(product_id = premium.id)],
+                    } for flavor in TasteByProduct.objects.select_related('taste').filter(product_id = premium.id)],
```

Comment:
```suggestion
                    } for flavor in premium.tastebyproduct_set.all()],
```
위의 코멘트와 현재 코멘트를 반영하면 기존 코드와 똑같이 동작할 것이고 orm 최적화 (select_related, prefetch_related) 적용은 14, 15 line에서 필요에 따라 한번에 작성해준다면 더 가독성 있으면서도 불필요한 쿼리의 개수를 줄일 수 있을 것 같습니다!

한번 직접 해보시고 fresh_products에도 반영해주세요!

Reply:
반영은 했는데 14,15번쨰  orm 최적화 (select_related, prefetch_related)는 좀더 공부 해봐야 적용할 수 있을것 같습니다. ㅜ